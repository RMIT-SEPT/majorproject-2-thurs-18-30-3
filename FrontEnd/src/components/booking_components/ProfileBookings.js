import React, {useEffect} from 'react'
import BookingCard from './BookingCard'

import '../../containers/App.css'

//Displays a user profile in a modal view

function ProfileBookings({change, bookingSet, deleteFunc}) {
  const [isEditing, setIsEditing] = React.useState(false);
  const [bookings, setBookings] = React.useState([]);

  useEffect(() => {
    if(typeof bookingSet !== 'undefined')
    {
      setBookings(bookingSet);
    }
  }, [bookingSet]);
  
  const switchEditing = () =>
  {
    setIsEditing(!isEditing);
  }

  const cards = bookings.map((book) => {
    return( <BookingCard booking = {book}/>)
  });

  const edits = bookings.map((book) => {
    return( <div className = 'profile-booking-row'><button onClick={deleteFunc(book.id)}> <i className="material-icons md-32"> delete </i> </button><BookingCard booking = {book}/></div>)
  });

  return (
    <div className="modal-wrapper">
      <main className="modalPane">
        <div className="paneHeader">
            <button className="closeButton" onClick={() => change()}>
              <i className="material-icons md-32"> close </i>
            </button>
                <h2>Your Bookings</h2>
            <button className="editButton" onClick={switchEditing}>
                {isEditing ? 'cancel' : 'delete'}
            </button>
        </div>
        {isEditing ? edits : cards}
      </main>
    </div>
  )
}
export default ProfileBookings
