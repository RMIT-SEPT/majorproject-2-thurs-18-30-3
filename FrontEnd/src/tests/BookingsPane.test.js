import React from 'react';

import ProfileBookings from '../components/booking_components/ProfileBookings';
import { StaticRouter } from 'react-router';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';
const dummyBookings = [{"serviceName": "Test","customerId": 1,"employeeId": "2","time": "11:11","date": "1-1-1","id": 1}];
const changeFuncMock = jest.fn();
const deleteFuncMock = jest.fn();

const setup = () => {
    return render(
        <StaticRouter>
            <ProfileBookings change={changeFuncMock} bookingSet = {dummyBookings} deleteFunc={deleteFuncMock}/>
        </StaticRouter>
    )
  }

  //snapshot test
test('Test Pane Renders', () => {
    const pane = renderer.create(
        <StaticRouter>
            <ProfileBookings change={changeFuncMock} bookingSet = {dummyBookings} deleteFunc={deleteFuncMock}/>
        </StaticRouter>,
      ).toJSON();
      expect(pane).toMatchSnapshot();
});

test('Test Pane Contents', () => {
    const container = setup();
    
    expect(container.getAllByRole('button').length).toBe(2);
    expect(container.getAllByRole('heading')[0]).toHaveTextContent('Your Bookings');
    expect(container.getByRole('cell')).toHaveAttribute('class','booking-card');
});

test('Test Booking Card Contents', () => {
    const container = setup();
    
    expect(container.getByRole('cell')).toHaveTextContent('11:11');
    expect(container.getByRole('cell')).toHaveTextContent('1-1-1');
    expect(container.getByRole('cell')).toHaveTextContent('Test');
});

test('Test Button functions', () => {
    const container = setup();
    
    const closeBtn= container.getAllByRole('button')[0];
    const deleteBtn= container.getAllByRole('button')[1];

    fireEvent.click(closeBtn);
    expect(changeFuncMock).toHaveBeenCalled();

    fireEvent.click(deleteBtn);
    expect(deleteFuncMock).toHaveBeenCalled();
});