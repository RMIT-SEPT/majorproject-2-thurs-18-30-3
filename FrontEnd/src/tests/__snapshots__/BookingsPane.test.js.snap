// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Pane Renders 1`] = `
<div
  className="modal-wrapper"
>
  <main
    className="modalPane"
  >
    <div
      className="paneHeader"
    >
      <button
        className="closeButton"
        onClick={[Function]}
      >
        <i
          className="material-icons md-32"
        >
           close 
        </i>
      </button>
      <h2>
        Your Bookings
      </h2>
      <button
        className="editButton"
        onClick={[Function]}
      >
        delete
      </button>
    </div>
  </main>
</div>
`;
